### ISSUE TITLE
Error When Querying Attached DB

### ISSUE THREAD

**Issue by caseycrogers** [tekartik/sqflite#699]
If you attach a db and then attempt to query a table from the attached db, SQLFlite throws an error that it could not find the table.

### Repro
Bug tested on Android. Given `foo.db` containing table `foo` and `bar.db` containing `bar`:
```dart
var db = await openDatabase('foo.db');

// Attach bar.
await db.execute('''ATTACH DATABASE 'bar.db' AS bar_db'');

// Verify that bar was successfully attached.
print(await db.rawQuery('''PRAGMA database_list;'''));

// Query bar.
print(await db.rawQuery('''SELECT * FROM bar_db.bar;'''));
```

Console output:
```
[{seq: 0, name: main, file: <PATH TO FOO.DB>}, {seq: 2, name: bar_db, file: <PATH TO BAR.DB>}]
(1) no such table: bar_db.bar in "SELECT * FROM bar_db.bar;"".
.
.
.
```

Code can be verified sane by running it from a database connection in your favorite DB browser/SQLite CLI. Tested in Intellij's DB browser on Windows.

### Root Cause
I think what's happening is that `rawQuery` is explicitly specifying the db via `baseDatabaseMethodArguments` so the query can only access tables in the id. See: https://github.com/tekartik/sqflite/blob/641959e5f18cc863c14d42d60384aade83f113c6/sqflite_common/lib/src/database_mixin.dart#L407
https://github.com/tekartik/sqflite/blob/641959e5f18cc863c14d42d60384aade83f113c6/sqflite_common/lib/src/database_mixin.dart#L261

### Proposed Fix
I'm in way over my head-don't really know how any of this works. but maybe you could add an optional argument to the relevant query methods that let developers specify the ID? In particular, it should let them specify a null ID such that users can write queries that span the main db and attached dbs:
```sql
SELECT *
FROM bar_db.bar JOIN foo
ON (bar_db.bar.id = foo.id);
```

**Comment by alextekartik** [comment]
Thanks for the report. Indeed that is something that I have not tested at all as this is not a very common usage on mobile. On the native side we only have a single database handle which is abstracted by an id on the dart side.

You might want to try attaching the database with a full path.

I just tried the following on Android with success:

```dart
var dbsPath = await databaseFactory.getDatabasesPath();
var db1Path = join(dbsPath, 'db1');
var db2Path = join(dbsPath, 'db2');

// Delete both database
await databaseFactory.deleteDatabase(db1Path);
await databaseFactory.deleteDatabase(db2Path);

// Create some data on db1 and close it
var db1 = await databaseFactory.openDatabase(db1Path);
var batch = db1.batch();
batch.execute('CREATE TABLE table1 (col1 INTEGER)');
batch.insert('table1', {'col1': 1234});
await batch.commit();
await db1.close();

// Open a new db2 database, attach db1 and query it
var db2 = await databaseFactory.openDatabase(db2Path);
await db2.execute('ATTACH DATABASE \'$db1Path\' AS db1');
var rows = await db2.query('db1.table1');
print(rows); // prints [{col1: 1234}]
```

**Comment by caseycrogers** [comment]
Hmm your code worked for me and then I fiddled with some other parts of my code and now mine is working too-I'm not 100% sure what was broken on my end but I fixed it! I think `bar.db` might have been corrupt? Either way, there's no bug on SQFLite's end! Closing this.

Also, attach is crazy useful in some narrow contexts! My database consists of one table that's prepopulated and never written to and a second table that starts empty and saves entries that the user has "bookmarked". When I push an update that adds new items to the pre-populated table, I can just delete the file and replace it with the new one and the user will retain their bookmarks.
